src/BatteryStateOfChargeService.h

```diff
@@ -18,5 +18,22 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
-    // Recommend adding necessary variables here
+    double InitialCurrent;
+    double CurrentPrime = 0;
+    double AmpHours;
+    double AmpHChange;
+    double AverageCurrent;
+    QTime InitialTime;
+    QTime TimePrime;
+    QTime t;
+    double TimeLeft;
```

Comment:
In the header file it is best to only have the variables that state you need keep track of between function calls. (Think of them like global variables that apply to BatteryStateOfChargeService). Some of these don't need to be kept track of every time addData is called. For example TimeLeft could just be created each time it is used within addData.

As a general rule it is best to have each variable have the smallest scope as possible. That means that it is only defined for as long as it is used.

Move TimeLeft, TimeChange, HoursPrime MinutesPrime, SecondsPrime, AverageCurrentPrime, InitialCurrent, AmpHChange, InitialTime to just be created within addData.

Also delete NewCurrent as it is not used.